suite-native/module-connect-device/src/components/PinOnDevice.tsx

```diff
@@ -0,0 +1,79 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { Dimensions } from 'react-native';
+import { useCallback, useEffect } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { Image, Box, Text } from '@suite-native/atoms';
+import { DeviceModelInternal } from '@trezor/connect';
+import {
+    removeButtonRequests,
+    selectDevice,
+    selectDeviceModel,
+    selectDeviceRequestedPin,
+} from '@suite-common/wallet-core';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import TrezorConnect, { UI_REQUEST } from '@trezor/connect';
+
+const SCREEN_HEIGHT = Dimensions.get('screen').height;
+
+export const deviceImageMap = {
+    [DeviceModelInternal.T1B1]: require('../assets/pin-t1b1.png'),
+    [DeviceModelInternal.T2T1]: require('../assets/pin-t2t1.png'),
+    [DeviceModelInternal.T2B1]: require('../assets/pin-t2b1.png'),
+} as const satisfies Record<DeviceModelInternal, string>;
+
+const wrapperStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingTop: utils.spacings.large,
+}));
+const imageStyle = prepareNativeStyle(_ => ({
+    maxHeight: SCREEN_HEIGHT * 0.6,
+    width: 243,
+    height: 550,
+    alignItems: 'center',
+    contentFit: 'contain',
+}));
+
+export const PinOnDevice = () => {
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
+
+    const device = useSelector(selectDevice) ?? null;
+    const deviceModel = useSelector(selectDeviceModel);
+    const hasDeviceRequestedPin = useSelector(selectDeviceRequestedPin);
+    const { applyStyle } = useNativeStyles();
+
+    const handleSuccess = useCallback(() => {
+        if (navigation.canGoBack()) {
+            navigation.goBack();
+        }
+        dispatch(removeButtonRequests({ device, buttonRequestCode: 'ButtonRequest_PinEntry' }));
+    }, [navigation, dispatch, device]);
+
+    useEffect(() => {
+        if (!hasDeviceRequestedPin) {
+            handleSuccess();
+        }
+    }, [hasDeviceRequestedPin, handleSuccess]);
```

Comment:
Just question, how can you get into state that would trigger this effect?

Reply:
`hasDeviceRequestedPin` is false here when TT/TS3 gets locked after you enter correct PIN again and confirm (e.g. when you try to verify receive address).

Reply:
nit: maybe it's worth a comment? It's not that obvious from the code

Reply:
391f48390c